### PR TITLE
Ruff format

### PR DESCRIPTION
**Pull request by purva-thakre**
## Description
Ruff workflow job might not be working as intended in https://github.com/vprusso/toqito/blob/b58f066e96fadb73364554b23ccb808a11f3c90a/.github/workflows/build-test-actions.yml#L27

https://github.com/ChartBoost/ruff-action?tab=readme-ov-file#ruff-action

Running `ruff format .` in `toqito` altered 114 files in total for the branch in #509 . 

## Changes
Notable changes that this PR has either accomplished or will accomplish. Feel free to add more lines to the itemized list
below.

  -  [ ] Change 1

## Checklist
Before marking your PR ready for review, make sure you checked the following locally. If this is your first PR, you might be notified of some workflow failures after a maintainer has approved the workflow jobs to be run on your PR. 

Additional information is available in the [documentation](https://toqito.readthedocs.io/en/latest/contributing.html#testing).

  -  [ ] Use `ruff` and `pylint` for errors related to code style and formatting.
  -  [ ] Verify all previous and newly added unit tests pass in `pytest`.
  -  [ ] Check the documentation build does not lead to any failures. `Sphinx` build can be checked locally for any failures related to your PR
  -  [ ] Use `linkcheck` to check for broken links in the documentation
  -  [ ] Use `doctest` to verify the examples in the function docstrings work as expected.

### CODE REVIEW

**Comment by purva-thakre** [comment]
Why does `ruff format .` change 114 files but `ruff check .` does not identify any issue with the changes 114 files?

**Comment by purva-thakre** [comment]
Figured out my confusion related to `ruff check .` and `ruff format .`. 

The former is a linter for Flake8 (plus dozens of plugins), isort, pydocstyle, pyupgrade, autoflake, and more whereas the latter utilizes Black. 

https://docs.astral.sh/ruff/formatter/
https://docs.astral.sh/ruff/linter/

The `ruff` action in the Github actions job only takes care of the linter rules not black. 

@vprusso This is ready to be merged. LMK if I forgot to catch something here.

File: pyproject.toml [hunk 1]
```diff
@@ -87,7 +87,7 @@ ignore = ["D407", "D203", "D213", "D416", "PLR0912", "PLR0911", "PLR0915", "PLR2
 # PLR0915 -- Checks for functions or methods with too many statements. The default number is 50. 
 # PLR2004 -- Checks if an equality or an inequality is compared to a variable or a num (prefers variable pre)
 # PLR0913 Too many arguments in function definition
-
+# W1404 -- String literals are implicitly concatenated in a literal iterable definition : maybe a comma is missing ?
```
Comment (purva-thakre) [inline pyproject.toml:90]:
```suggestion

```

**Comment by vprusso** [comment]
This is great. Awesome to see all of the consistency updates. Went through the diff at a high-level, but everything looked pretty good and sane to me. Nice work, @purva-thakre !

**Review by vprusso — APPROVED** [review]
Awesome, looks great! MWR! 🚀

**Pull request merged**